Source/AppAuthTV/OIDTVAuthorizationService.m

```diff
@@ -60,73 +60,29 @@ + (void)discoverServiceConfigurationForIssuer:(NSURL *)issuerURL
 }
 
 + (void)discoverServiceConfigurationForDiscoveryURL:(NSURL *)discoveryURL
-    completion:(OIDTVDiscoveryCallback)completion {
-
-  NSURLSession *session = [OIDURLSessionProvider session];
-  NSURLSessionDataTask *task =
-      [session dataTaskWithURL:discoveryURL
-             completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
-    // If we got any sort of error, just report it.
-    if (error || !data) {
-      NSString *errorDescription =
-          [NSString stringWithFormat:@"Connection error fetching discovery document '%@': %@.",
-                                     discoveryURL,
-                                     error.localizedDescription];
-      error = [OIDErrorUtilities errorWithCode:OIDErrorCodeNetworkError
-                               underlyingError:error
-                                   description:errorDescription];
-      dispatch_async(dispatch_get_main_queue(), ^{
-        completion(nil, error);
-      });
-      return;
-    }
+                                         completion:(OIDTVDiscoveryCallback)completion {
+  // Call the corresponding discovery method in OIDAuthorizationService
+  [OIDAuthorizationService discoverServiceConfigurationForDiscoveryURL:discoveryURL
+      completion:^(OIDServiceConfiguration * _Nullable configuration, NSError * _Nullable error) {
+    if (configuration) {
```

Comment:
Instead of nesting if statements, try to have early exits/returns.

e.g.
```
if (configuration == nil) {
  completion(nil, error);
  return;
}

if (configuration.discoveryDocument.deviceAuthorizationEndpoint == nil {
  // rest of the code
```